R/FilteredData-utils.R

```diff
@@ -32,50 +32,32 @@ init_filtered_data.TealData <- function(x, # nolint
                                         code = x$get_code_class(),
                                         cdisc = FALSE,
                                         check = x$get_check()) {
+  cdisc <- if (length(join_keys$get_parents()) > 0) TRUE else FALSE
   data_objects <- lapply(x$get_datanames(), function(dataname) {
     dataset <- x$get_dataset(dataname)
-    list(
-      dataset = dataset$get_raw_data(),
-      keys = dataset$get_keys(),
-      metadata = dataset$get_metadata(),
-      label = dataset$get_dataset_label()
-    )
-  })
-  names(data_objects) <- x$get_datanames()
-
-  init_filtered_data(
-    x = data_objects,
-    join_keys = join_keys,
-    code = code,
-    check = check
-  )
-}
 
-#' @keywords internal
-#' @export
-init_filtered_data.CDISCTealData <- function(x, # nolint
-                                             join_keys = x$get_join_keys(),
-                                             code = x$get_code_class(),
-                                             cdisc = TRUE,
-                                             check = x$get_check()) {
-  data_objects <- lapply(x$get_datanames(), function(dataname) {
-    dataset <- x$get_dataset(dataname)
-
-    # CDISCTealData can contain TealDataset and CDISCTealDataset objects
-    # the former do not have a get_parent() call
     parent <- if (inherits(dataset, "CDISCTealDataset")) {
       dataset$get_parent()
     } else {
       NULL
     }
 
-    list(
-      dataset = dataset$get_raw_data(),
-      keys = dataset$get_keys(),
-      metadata = dataset$get_metadata(),
-      label = dataset$get_dataset_label(),
-      parent = parent
-    )
+    if (cdisc) {
```

Comment:
I think you can make this nicer by doing:

sensible_name <- list(dataset = ..., label = ...)
if (cdisc) sensible_name[["parent"]] <- parent

Or - if it's ok for parent to be NULL as opposed to not be in names(list) for non cdisc case:

list(
  dataset = ....,
  ...
  parent = if (cdisc) parent else NULL
)

(or if parent is NULL in non-cdisc case then it's even simpler)

Reply:
updated.